ocean_lib/ocean/test/test_ocean_assets.py

```diff
@@ -38,43 +41,145 @@ def test_register_asset(publisher_ocean_instance, publisher_wallet, consumer_wal
     assert ocn.assets.resolve(invalid_did) is None
 
 
-@pytest.mark.skip(reason="TODO: update function on OceanAssets class")
-def test_update(publisher_ocean_instance, publisher_wallet, consumer_wallet, config):
-    ocn = publisher_ocean_instance
-    block = ocn.web3.eth.block_number
-    alice = publisher_wallet
-    bob = consumer_wallet
+def test_update_metadata(publisher_ocean_instance, publisher_wallet, config):
+
+    ddo = create_asset(publisher_ocean_instance, publisher_wallet, config)
+
+    new_metadata = copy.deepcopy(ddo.metadata)
+
+    # Update only metadata
+    _description = "Updated description"
+    new_metadata["description"] = _description
+    new_metadata["updated"] = datetime.now().isoformat()
+    ddo.metadata = new_metadata
+
+    _asset = publisher_ocean_instance.assets.update(
+        asset=ddo,
+        publisher_wallet=publisher_wallet,
+    )
+
+    assert _asset.datatokens == ddo.datatokens
+    assert len(_asset.services) == len(ddo.services)
+    assert _asset.services[0].as_dictionary() == ddo.services[0].as_dictionary()
+    assert _asset.credentials == ddo.credentials
```

Comment:
ok let's assert that the updated values truly corresponds ("updated" and "description" keys in this case)